### PR TITLE
ready for merge request

### PR DESCRIPTION
**Pull request by natemella**
#### What scientific question is your analysis addressing?
image

#### What was your approach?
image

#### What GitHub issue does your pull request address?
image

#### Which areas should receive a particularly close look?
image

#### Is there anything that you want to discuss further?
image

#### What types of results are included (e.g., table, figure)?
image

#### What is your summary of the results?

We tested for batch effects in three different areas:
1.	Batch = Sequencing Center
2.	Batch = Cohort
3.	Batch = polyA vs ribodepletion

Within each of these analyses I evaluated RNA seq data that had been prepared using either RSEM normalization or Kallisto (tpm) values.
The null hypothesis is that there are no batch effects in the data

|          |               | Batch                                          |                                          |                                    |
|----------|---------------|-------------------------------------------------------|------------------------------------------|------------------------------------|
|          |               | **Sequence center**                           | **Cohort**                               | **Preparation method**                 |
| RSEM     | polyA         | **reject**                                      | **reject**                                   | **reject**                             |
|          |               | before                                        | before                                   | before                             |
|          |               | - [rsem_polyA_sequence.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234696/rsem_polyA_sequence.pdf)                   | - [rsem_polya_cohort_pca.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234692/rsem_polya_cohort_pca.pdf)            | - [rsem_method.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234686/rsem_method.pdf)               |
|          |               | - [rsem_polyA_sequence_pca.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234687/rsem_polyA_sequence_pca.pdf)               | - [rsem_polyA_cohort.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234697/rsem_polyA_cohort.pdf)                | - [rsem_method_pca.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234691/rsem_method_pca.pdf)            |
|          |               | after                                         | after                                    | after                              |
|          |               | - [rsem_polya_sequence_combat.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4325812/rsem_polya_sequence_combat.pdf)            | - [rsem_polya_cohort_combat_pca.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4325813/rsem_polya_cohort_combat_pca.pdf)     | - [rsem_method_combat.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4325817/rsem_method_combat.pdf)        |
|          |               | - [rsem_polya_sequence_combat_pca.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4325811/rsem_polya_sequence_combat_pca.pdf)        | - [rsem_polya_cohort_combat.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4325814/rsem_polya_cohort_combat.pdf)         | - [rsem_method_combat_pca.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4325815/rsem_method_combat_pca.pdf)    |
|          |               |                                               |                                          |                                    |
|          | ribodepletion | **reject**                                        | **fail to reject**                           |                                    |
|          |               | before                                        | before                                   |                                    |
|          |               | - [rsem_stranded_sequence.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234683/rsem_stranded_sequence.pdf)               | - [rsem_stranded_cohort.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234695/rsem_stranded_cohort.pdf)             |                                    |
|          |               | - [rsem_stranded_sequence_pca.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234681/rsem_stranded_sequence_pca.pdf)           | - [rsem_stranded_cohort_pca.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234693/rsem_stranded_cohort_pca.pdf)         |                                    |
|          |               | after                                         | after                                    |                                    |
|          |               | - [rsem_stranded_sequence_combat_pca.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4325809/rsem_stranded_sequence_combat_pca.pdf)     |                                    |                                    |
|          |               | - [rsem_stranded_sequence_combat.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4325810/rsem_stranded_sequence_combat.pdf)         |                                    |                                    |
|          |               |                                               |                                          |                                    |
| Kallisto | polyA         | **reject**                                        | **reject**                                   | **reject**                             |
|          |               | before                                        | before                                   | before                             |
|          |               | - [kallisto_polyA_sequence.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234682/kallisto_polyA_sequence.pdf)               | - [kallisto_polyA_cohort.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234684/kallisto_polyA_cohort.pdf)            | - [kallisto_method_pca.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234694/kallisto_method_pca.pdf)        |
|          |               | - [kallisto_polya_sequence_pca.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234680/kallisto_polya_sequence_pca.pdf)           | - [kallisto_polyA_cohort_pca.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234685/kallisto_polyA_cohort_pca.pdf)        | - [kallisto_method.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234688/kallisto_method.pdf)            |
|          |               | after                                         | after                                    | after                              |
|          |               | - [kallisto_polya_sequence_combat_pca.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4325819/kallisto_polya_sequence_combat_pca.pdf)    | - [kallisto_polya_cohort_combat.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4328136/kallisto_polya_cohort_combat.pdf)     | - [kallisto_method_combat.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4325822/kallisto_method_combat.pdf)    |
|          |               | - [kallisto_polya_sequence_combat.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4325820/kallisto_polya_sequence_combat.pdf)        | - [kallisto_polya_cohort_combat_pca.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4328137/kallisto_polya_cohort_combat_pca.pdf) | - [kallisto_method_combat_pca.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4325821/kallisto_method_combat_pca.pdf) |
|          | ribodepletion | **reject**                                        | **fail to reject**                           |                                    |
|          |               | before                                        | before                                   |                                    |
|          |               | - [kallisto_stranded_sequence_pca.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234699/kallisto_stranded_sequence_pca.pdf)        | - [kallisto_stranded_cohort_pca.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234689/kallisto_stranded_cohort_pca.pdf )    |                                    |
|          |               | - [kallisto_stranded_sequence.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234690/Kallisto_stranded_sequence.pdf)            | - [kallisto_stranded_cohort.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234698/kallisto_stranded_cohort.pdf)        |                                    |
|          |               | after                                         | after                                    |                                    |
|          |               | - [kallisto_stranded_sequence_combat.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4325818/kallisto_stranded_sequence_combat.pdf)      |                                    |                                    |
|          |               | - [kallisto_stranded_sequence_combat_pca.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4325816/kallisto_stranded_sequence_combat_pca.pdf)      |                                   |                                    |
|          |               |                                               |                                          |                                    | 

After running combat, it appears that the batch effects are successfully eliminated. Attached are the two graphs for each section 1) the distribution of P-values and 2) the PCA plot generated from batchQC

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by cgreene** [comment]
Hi @natemella - if you could paste text instead of images I'd quote tweet, but your answer to this one remains an ongoing question:
> Is there anything that you want to discuss further?

You've batch effect corrected for different datasets, but we suspect those may be confounded with disease type. this means your attempts to remove unwanted variability might actually induce other confounded variability. If you wanted to check this, you'd pick one or a few dominant histologies from those centers, compare them, and at least see that:
1. Those histologies now overlap, or at least overlap better than they did before.
2. The histologies that are not those within the center(s) in question retain roughly the same variability, looking within center, that they did before the data were merged.

Without doing this analysis, I don't think there's any way to know if this correction is helping or hurting.

**Comment by jaclyn-taroni** [comment]
I am going to resolve conflicts and add this analysis to continuous integration.

File: analyses/batch-effects/run_batch_effects.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+#SBATCH --time=02:00:00   # walltime
+#SBATCH --ntasks=1   # number of processor cores (i.e. tasks)
+#SBATCH --nodes=1   # number of nodes
+#SBATCH -C 'rhel7'   # features syntax (use quotes): -C 'a&b&c&d'
+#SBATCH --mem=120G   # memory 
+
```
[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply analyses/batch-effects/run_batch_effects.sh:8]: For CI, we necessarily need to run things from the root of the repo - so this is called via
```sh
bash analyses/batch-effects/run_batch_effects.sh
```
We get the following error (which you can see [here](
https://circleci.com/gh/AlexsLemonade/OpenPBTA-analysis/4066)):

```
^@^@Fatal error: cannot open file 'batch-sequence-effects.R': No such file or directory
Fatal error: cannot open file 'batch-cohort-effects.R': No such file or directory
Fatal error: cannot open file 'batch-method-effects.R': No such file or directory

Exited with code exit status 2
```

Adding the one liner that sets the working directory to the directory of this file should solve it!

**Comment by jaclyn-taroni** [comment]
On the connected issue, the next steps included new pull requests https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/448#issuecomment-617364339 - the first of which was edits to the Docker image. The Docker image has been overhauled (#689) since this pull request was opened, so I think the best course of action is to close this one at this point.

File: analyses/batch-effects/README.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+|          |               | - [kallisto_stranded_sequence.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234690/Kallisto_stranded_sequence.pdf)            | - [kallisto_stranded_cohort.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4234698/kallisto_stranded_cohort.pdf)        |                                    |
+|          |               | after                                         | after                                    |                                    |
+|          |               | - [kallisto_stranded_sequence_combat.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4325818/kallisto_stranded_sequence_combat.pdf)      |                                    |                                    |
+|          |               | - [kallisto_stranded_sequence_combat_pca.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/4325816/kallisto_stranded_sequence_combat_pca.pdf)      |                                   |                                    |
+|          |               |                                               |                                          |                                    |
```
[the remark this answers is not in the document]
Comment (natemella) [reply analyses/batch-effects/README.md:68]: This is where I was storing my graphs

File: analyses/batch-effects/batch-cohort-effects.R [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    tibble::enframe(name = NULL) %>%
+    tidyr::separate(col=value, into=c("key", "value"), sep="=", fill='right') %>%
+    dplyr::filter(key == "--file") %>%
+    dplyr::pull(value)
+  if (length(this_file)==0)
+  {
+    this_file <- rstudioapi::getSourceEditorContext()$path
+  }
+  return(dirname(this_file))
+}
```
[the remark this answers is not in the document]
Comment (natemella) [reply analyses/batch-effects/batch-cohort-effects.R:18]: This looks like an outdated function

File: analyses/batch-effects/run_batch_effects.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (natemella) [reply analyses/batch-effects/run_batch_effects.sh:8]: Thanks for these tips

File: analyses/batch-effects/batch-cohort-effects.R [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  poly1matrix = poly1matrix[which((apply(poly1matrix, 1, var)>0)),]
+
+  # log tranform data so batchQC and combat run properly
+  poly1matrix = log2(poly1matrix + 1)
+
+  print("DATA IS READY") 
+  
+  # run batchQC and only generate the P-value distribution and PCA plot
+  batchQC(poly1matrix, batch=batch,
+          report_file=report_name, report_dir=output_dir,
```
[the remark this answers is not in the document]
Comment (natemella) [reply analyses/batch-effects/batch-cohort-effects.R:80]: Yes it will

**Pull request closed without merging**